wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -1622,14 +1627,19 @@ protected JSch getJSch(final OpenSshConfig.Host hc, FS fs) throws JSchException
                     String password = remoteUrl.substring(remoteUrl.lastIndexOf(":") + 1, remoteUrl.indexOf("@"));
                     command.setCredentialsProvider(new UsernamePasswordCredentialsProvider(username, password));
                 } else {
+                    String gitUser = request.getAttributes().get("userName");
+                    String gitPassword = request.getAttributes().get("password");
+                    if (gitUser != null && gitPassword != null) {
+                        setCurrentCredentials(gitUser, gitPassword);
+                        requestCredentialsHaveBeenUsed = true;
+                    }
                     credentials = credentialsLoader.getUserCredential(remoteUrl);
                     if (credentials != null) {
-                        command.setCredentialsProvider(new UsernamePasswordCredentialsProvider(credentials.getUserName(),
-                                                                                               credentials.getPassword()));
+                        command.setCredentialsProvider(
```

Comment:
@sleshchenko  - The first check is for SSH and then for HTTP.
It check first the https?://[^:]+:[^@]+@.* and then contains credentials. DId you meant to change the order of the checks ? in the https I think they want that the user and password will be in the path and it will not occurred on the second case